Add documentation for Loki logging stack (#2)
appuio/managed-openshift-docs: docs/modules/ROOT/pages/logging/tutorial-lokistack.adoc

> = Tutorial: Access application logs with Lokistack

Reviewer: ```suggestion
= Getting started with logging: Lokistack
```

This tutorial explains how to view the logs of an application on Openshift.

== Requirements

To follow this guide, please make sure that you have the following tools installed:

> `oc`:: You can download the OpenShift command directly from {product}, selecting the help menu (marked as a question mark) and selecting the "Command line tools" entry

Reviewer: ```suggestion
`oc`:: You can download the OpenShift command directly from OpenShift Web Console. Open the help menu (marked as a question mark at the top right) and select the "Command line tools" entry.
```

== Step 1: Deploy Fortune application

> . Create a new namespace called `vshn-demo-fortune`

Reviewer: ```suggestion
. Create a new namespace called `demo-fortune`
```

I second @simu. Plus, it is not consistent with the below shell snipped anyway.

+
[source,shell]
--
oc create ns demo-fortune
--
+
[TIP]
====
If you get an error that the namespace already exists, please pick another name and try again.
We recommend prefixing the name with the name of your organization to minimize potential for name collisions.
====

. Create Fortune application manifest
+
.Manifest for Fortune application
[source,yaml]
--
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: fortune-cookie
  name: fortune-cookie
spec:
  replicas: 1
  selector:
    matchLabels:
      app: fortune-cookie
  template:
    metadata:
      labels:
        app: fortune-cookie
    spec:
      containers:
      - image: registry.gitlab.com/vshn/applications/fortune-go:latest
        imagePullPolicy: IfNotPresent
        name: fortune-cookie
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: fortune-cookie
  name: fortune-cookie
spec:
  ports:
  - port: 8080
    protocol: TCP
    targetPort: 8080
    name: web
  selector:
    app: fortune-cookie
  type: ClusterIP
--

. Deploy Fortune application manifest
+
[source,shell]
--

> oc -n demo-fortune apply -f fortune.yaml

Reviewer: ```suggestion
oc apply -f fortune.yaml
```

This is a tutorial where the commands are meant to be executed in the given order. Previously, we used `oc create namespace`. If we used `oc create project` instead, the current context would already be set appropriately.

--

. Expose Fortune application
+
[source,shell]
--
oc -n demo-fortune expose svc/fortune-cookie

> oc -n demo-fortune patch route fortune-cookie -p '{"spec":{"tls":{"termination":"edge","insecureEdgeTerminationPolicy":"Redirect"}}}'

Reviewer: Is this relevant for showcasing the subject at hand? Tutorials should focus on showcasing one thing. Mixing in not related stuff, might be confusing.

Author: Somewhat yes, you can create log entries by accessing the webpage.

Reviewer: But is it relevant if is accessed via HTTPS, or would HTTP also work? I know, security should always be a thing, but I would prefer to have this aspect explained in a dedicated section about ingress.

= Tutorial: Access application logs with Lokistack

This tutorial explains how to view the logs of an application on Openshift.

== Requirements

To follow this guide, please make sure that you have the following tools installed:

`oc`:: You can download the OpenShift command directly from {product}, selecting the help menu (marked as a question mark) and selecting the "Command line tools" entry

== Step 1: Deploy Fortune application

> . Create a new namespace called `vshn-demo-fortune`

Reviewer 2: I'd prefer to not use a namespace prefixed with `vshn-`. While there's no policies in place to prevent arbitrary users from using the `vshn-` prefix, I don't think it's a particularly good idea to use the prefix in a tutorial.

+
[source,shell]
--
oc create ns vshn-demo-fortune
--

> 

Reviewer 2: You probably want something along the lines of

```suggestion

[TIP]
====
If you get an error that the namespace already exists, please pick another name and try again.
We recommend prefixing the name with the name of your organization to minimize potential for name collisions.
====
```

. Create Fortune application manifest
+
.Manifest for Fortune application
[source,yaml]
--
apiVersion: apps/v1
kind: Deployment
metadata:
  labels:
    app: fortune-cookie
  name: fortune-cookie
spec:
  replicas: 1
  selector:
    matchLabels:
      app: fortune-cookie
  template:
    metadata:
      labels:
        app: fortune-cookie
    spec:
      containers:
      - image: registry.gitlab.com/vshn/applications/fortune-go:latest
        imagePullPolicy: IfNotPresent
        name: fortune-cookie
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: fortune-cookie
  name: fortune-cookie
spec:
  ports:
  - port: 8080
    protocol: TCP
    targetPort: 8080
    name: web
  selector:
    app: fortune-cookie
  type: ClusterIP
---
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  annotations:
    cert-manager.io/cluster-issuer: letsencrypt-production
  name: fortune-cookie
spec:
  rules:
  - host: <INSERT_URL_FOR_FORTUNE_APP>
    http:
      paths:
      - pathType: Prefix
        path: /
        backend:
          service:
            name: fortune-cookie
            port:
              number: 8080
  tls:
  - hosts:
    - <INSERT_URL_FOR_FORTUNE_APP>

>     secretName: fortune-cookie-tls

Reviewer 2: I'd consider leaving out the ingress, and instead using `oc expose svc/fortune-cookie` instead for the tutorial, something like

```
oc -n vshn-demo-fortune expose svc/fortune-go
oc -n vshn-demo-fortune patch route fortune-go -p '{"spec":{"tls":{"termination":"edge","insecureEdgeTerminationPolicy":"Redirect"}}}'

```
